src/test/scala/higherkindness/compendium/core/ProtocolUtilsSpec.scala

```diff
@@ -62,27 +63,97 @@ object ProtocolUtilsSpec extends Specification with ScalaCheck {
       val stream: InputStream = getClass.getResourceAsStream("/correct.proto")
       val text: String        = scala.io.Source.fromInputStream(stream).getLines.mkString
       val protocol: Protocol  = Protocol(text)
+      stream.close()
+
       validator.validateProtocol(protocol, IdlName.Protobuf).unsafeRunSync === protocol
     }
 
-    "[Proto] Given a raw protocol text raises an error if the protocol is incorrect" >> {
+    "[Proto] Given a raw protocol text raises an error if idl specified is incorrect" >> {
       val stream: InputStream = getClass.getResourceAsStream("/correct.avro")
       val text: String        = scala.io.Source.fromInputStream(stream).getLines.mkString
       val protocol: Protocol  = Protocol(text)
+      stream.close()
+
       validator
         .validateProtocol(protocol, IdlName.Protobuf)
         .unsafeRunSync must throwA[SchemaParseException]
     }
 
-    "[Proto] Given a raw protocol text raises an error if the protocol is incorrect II" >> {
+    "[Proto] Given a raw protocol text raises an error if the protocol is incorrect" >> {
       val stream: InputStream = getClass.getResourceAsStream("/wrong.proto")
       val text: String        = scala.io.Source.fromInputStream(stream).getLines.mkString
       val protocol: Protocol  = Protocol(text)
+      stream.close()
+
       validator
         .validateProtocol(protocol, IdlName.Protobuf)
         .unsafeRunSync must throwA[SchemaParseException]
     }
 
-  }
+    "[OpenAPI - Yaml] Given a raw protocol text returns the same protocol if it validates correctly" >> {
+      val stream: InputStream = getClass.getResourceAsStream("/correctYamlOpenAPI.yaml")
+      val text: String        = scala.io.Source.fromInputStream(stream).getLines.mkString("\n")
+      val protocol: Protocol  = Protocol(text)
+      stream.close()
+
+      validator
+        .validateProtocol(protocol, IdlName.OpenAPI)
+        .unsafeRunSync === protocol
+    }
+
+    "[OpenAPI - Yaml] Given a raw protocol text raises an error if the protocol is incorrect" >> {
+      val stream: InputStream = getClass.getResourceAsStream("/wrongYamlOpenAPI.yaml")
+      val text: String        = scala.io.Source.fromInputStream(stream).getLines.mkString("\n")
+      val protocol: Protocol  = Protocol(text)
+      stream.close()
```

Comment:
**suggestion**: use [Resource](https://typelevel.org/cats-effect/datatypes/resource.html#with-scalaiosource).

It doesn't need to be part of this PR :)

Reply:
Good idea!, but as you said I'd address it in a following PR

Reply:
Sure, thanks @juanpedromoreno! We'll have this in mind.